### DISCUSSION TITLE
HCP Pipeline

### DISCUSSION THREAD

**Discussion by spiritus01** [neurodesk/neurodesk.github.io-hugo#276] [Q&A] [answered]
Hello, 
I have enjoyed using Neurodesktop for 3 months.  Recently, I had to run the HCP pipeline (https://github.com/Washington-University/HCPpipelines) for several MRI projects, which was hard to do in Neurodesktop.  First, several prerequisites of the pipeline are not in Neurodesktop, including ...

- Connectome Workbench
- HCP version of gradunwarp
- MSM_HOCR v3.0/(github v1.0)
- FSL FIX v 1.0.6.14+

When I tried to install them in Neurodesktop, it was hard and I repeatedly got errors, which are hard to understand.  Is there any pre-compiled version of HCP pipeline available to Neurodesktop or any simple way to install the prerequisites?

**Comment by stebo85** [comment, marked as answer]
Dear @spiritus01 

Thank you so much for the feedback :) That's true - we currently don't have all these tools available inside the project. We do have the connectome workbanch and FSL (but maybe not FSL FIX if it's not included in the default FSL), but the HCP version of gradunwarp or MSM are not there as far as I know.

We can add these tools to Neurodesktop if you can help with testing :)

Are you aware if this BIDS app would already work for you? 
https://hub.docker.com/r/bids/hcppipelines

Thank you
Steffen

**Comment by spiritus01** [comment]
Dear Steffen,
Thank you for your reply and the information.  The BIDS app appears to be a
good option that I can use right now, but I would like to conduct all of my
MRI analysis in a single Docker environment, Neurodesktop.
I will be happy to help if there is anything that I can do for testing and
others to add the tools for the HCP pipeline. Let me know what you think.

Best regards,

- Steven

**Reply by stebo85** [reply]
Dear @spiritus01 - I added hcppipelines to Neurodesk. Once you restart your neurodesk instance you should find it in the Application menu under BIDS apps -> bidsappshcppipelines
<img width="1067" alt="image" src="https://github.com/NeuroDesk/neurodesk.github.io/assets/4021595/a05c2005-27d5-4f74-8f3f-cc90283dc074">

Can you let me know if this works and if you see any issues :)

**Comment by spiritus01** [comment]
That's great news!
Thank you for letting me know.  Due to a few urgent deadlines, I'll be able
to test it sometime later next week.  I'll let you know about my experience
with the new tool.

Best wishes,

- Steven
